pydantic_ai_slim/pydantic_ai/_tool_manager.py

```diff
@@ -156,7 +156,7 @@ async def _call_tool(
             else:
                 args_dict = validator.validate_python(call.args or {}, allow_partial=pyd_allow_partial)
 
-            result = await self.toolset.call_tool(name, args_dict, ctx, tool)
+            result = await self.toolset.call_tool(name, args_dict, ctx, tool, allow_partial=allow_partial)
```

Comment:
Since `AbstractToolset` is a public class that we encourage users to subclass, we unfortunately can't add an argument as that'd be a breaking change. I suggest passing this value through a new `tool_args_partial` field on `RunContext` instead.

Note that we'll also need to update `TemporalRunContext` (and the Temporal docs) to include that field by default.